torchtune/modules/model_fusion/_fusion.py

```diff
@@ -269,8 +271,8 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         # [batch_size x seq_length x embed_dim]
         out = self._fused_embed(bs, seq_len)
         mask = mask.unsqueeze(-1).expand(bs, seq_len, self.dim)
-        out.masked_scatter_(mask, embeds)
-        out.masked_scatter_(~mask, fusion_embeds)
+        out = out.masked_scatter(mask, embeds)
+        out = out.masked_scatter(~mask, fusion_embeds)
```

Comment:
Is this cause we need to track the grads?

Reply:
It's because using the inplace masked_scatter was giving bad gradients